sdk/client/src/macros.rs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+macro_rules! cfg_async {
+    ($($item:item)*) => {
+        $(
+            #[cfg(feature = "async")]
+            #[cfg_attr(docsrs, doc(cfg(feature = "async")))]
+            $item
+        )*
+    }
+}
+
+macro_rules! cfg_blocking {
+    ($($item:item)*) => {
+        $(
+            #[cfg(feature = "blocking")]
+            #[cfg_attr(docsrs, doc(cfg(feature = "blocking")))]
+            $item
+        )*
+    }
+}
+
+macro_rules! cfg_async_or_blocking {
```

Comment:
what is reason we need this macro? only expose structs when non of async / blocking feature is enabled?

Reply:
yeah exactly, there are some things that are shared between the two impls so they need to be enabled when either is enabled.